chia-keys/src/derive_keys.rs

```diff
@@ -0,0 +1,44 @@
+use crate::derivable_key::DerivableKey;
+use crate::secret_key::SecretKey;
+
+fn derive_path_unhardened<Key: DerivableKey>(key: &Key, path: &[u32]) -> Key {
+    let mut derived = key.derive_unhardened(path[0]);
+    for idx in &path[1..] {
+        derived = derived.derive_unhardened(*idx);
+    }
+    derived
+}
+
+fn derive_path_hardened(key: &SecretKey, path: &[u32]) -> SecretKey {
+    let mut derived = key.derive_hardened(path[0]);
+    for idx in &path[1..] {
+        derived = derived.derive_hardened(*idx);
+    }
+    derived
+}
+
+pub fn master_to_wallet_unhardened_intermediate<Key: DerivableKey>(key: &Key) -> Key {
+    derive_path_unhardened(key, &[12381_u32, 8444, 2])
```

Comment:
You might consider simply exporting these numbers (ie. [12381, 8444, 2]) from another function rather than hard-coding them inside a function. Call them `chia_default_wallet_path` or something more accurate.

Reply:
I don't see the upside of doing that. I want to keep the API as simple as possible, and as hard as possible to use the wrong derivation path. If we introduce another derivation path, I'd rather just add another function here that includes it.